### PR TITLE
LFX Spring 2023: meshery, tag-network, smp

### PR DESCRIPTION
**Pull request by leecalcote**
Signed-off-by: leecalcote <leecalcote@gmail.com>

### CODE REVIEW

File: lfx-mentorship/2023/01-Mar-May/README.md [hunk 1]
```diff
@@ -155,6 +163,21 @@ Mentee application instructions can be found on the [Program Guidelines](https:/
...
+#### Representing Kubernetes ontology in MeshModel
+
+- Description: Network topologies and graph databases go hand-in-hand. The OpenAPI specifications for Kubernetes provides taxonomy, but augmenting a graph data model with formalized ontologies enables any number of capabilities, one of the more straightforward is the inferencing requisite for natural language processing, and consequently, a human-centric query / response interaction becomes becomes possible. More importantly, more advanced systems can be built when a graph data model of connected systems is upgraded to be a knowledge semantic graph. Deliverables (among other items):
+
+- MeshModel capabilities browser
+- Import/export of MeshModel models and components as OCI images
+- augmentation of cuelang-based component generator
+
+- Recommended Skills: cuelang, golang, OCI
+- Mentor(s): [Lee Calcote](https://github.com/leecalcote)
```
Comment (nate-double-u) [inline lfx-mentorship/2023/01-Mar-May/README.md:178]:
```suggestion
- Mentor(s): Lee Calcote (@leecalcote, leecalcote@gmail.com)
````

File: lfx-mentorship/2023/01-Mar-May/README.md [hunk 2]
```diff
@@ -460,6 +483,29 @@ Note: This item is a work in progress. The selected mentee is expected to contin
 - Upstream Issue: https://github.com/kubeedge/ianvs/issues/48
 - LFX URL: https://mentorship.lfx.linuxfoundation.org/project/50cdbd65-e0cd-4c0f-8c63-6bd5c603ba89
 
+#### Meshery
+
+##### Distributed workflow engine
+
+- Description: Integrate a new architectural component into Meshery: a workflow engine. This project involves shifting Meshery off of bitcask and off of sqlite over to postgres using gorm (golang). Interns will familiarize with concepts of orchestration engines, including chaining workflows, and content lifecycle management.
+- Recommended Skills: Golang, Temporal, ReactJS
+- Mentor(s): Lee Calcote ([@lcalcote](https://twitter.com/lcalcote)),  Ashish Tiwari ([@revolyssup](https://twitter.com/revolyssup))
```
Comment (nate-double-u) [inline lfx-mentorship/2023/01-Mar-May/README.md:492]:
```suggestion
- Mentor(s): Lee Calcote (@leecalcote, leecalcote@gmail.com), Ashish Tiwari (ashishjaitiwari15112000@gmail.com)
```

**Review by nate-double-u — COMMENTED** [review]
Thanks for this @leecalcote, I've added email addresses for the mentors. Could you double check?

File: lfx-mentorship/2023/01-Mar-May/README.md [hunk 3]
```diff
@@ -155,6 +163,21 @@ Mentee application instructions can be found on the [Program Guidelines](https:/
...
+
+- Description: Network topologies and graph databases go hand-in-hand. The OpenAPI specifications for Kubernetes provides taxonomy, but augmenting a graph data model with formalized ontologies enables any number of capabilities, one of the more straightforward is the inferencing requisite for natural language processing, and consequently, a human-centric query / response interaction becomes becomes possible. More importantly, more advanced systems can be built when a graph data model of connected systems is upgraded to be a knowledge semantic graph. Deliverables (among other items):
+
+- MeshModel capabilities browser
+- Import/export of MeshModel models and components as OCI images
+- augmentation of cuelang-based component generator
+
+- Recommended Skills: cuelang, golang, OCI
+- Mentor(s): Lee Calcote (@leecalcote, leecalcote@gmail.com)
+- Issue: https://github.com/cncf/tag-network/issues/24
```
Comment (nate-double-u) [inline lfx-mentorship/2023/01-Mar-May/README.md:179]:
```suggestion
- Issue: https://github.com/cncf/tag-network/issues/24
- LFX URL: https://mentorship.lfx.linuxfoundation.org/project/96080e3d-83e2-46ed-928c-b6e7f3154bf3
```

File: lfx-mentorship/2023/01-Mar-May/README.md [hunk 4]
```diff
@@ -563,6 +609,18 @@ each subset of UnitedDeployment can have different metadata, container configura
...
+
+#### Adaptive Load Controller
+
+- Description: The adaptive load controller is to execute optimization routines recursivley to determine the maximum load a system can sustain. The maximum load is usually defined by the maximum requests per second (rps) the system can handle. The metrics (CPU usage, latency etc) collected from the system under test are the constraints we provide to judge whether a system under test (SUT) is sustaining the load.
+
+A use-case that fits very well is be the ability to use it to run performance tests on a schedule and track the maximum load a system can handle over time. This could give insights to performance improvements or degradations.
+
+- Recommended Skills: golang, grpc, docker, kubernetes
+- Mentor(s): Lee Calcote (@leecalcote, leecalcote@gmail.com), Xin Huang (@gyohuangxin, xin1.huang@intel.com)
+- Upstream Issue (URL): https://github.com/service-mesh-performance/service-mesh-performance/issues/350
```
Comment (nate-double-u) [inline lfx-mentorship/2023/01-Mar-May/README.md:622]:
```suggestion
- Upstream Issue (URL): https://github.com/service-mesh-performance/service-mesh-performance/issues/350
LFX URL: https://mentorship.lfx.linuxfoundation.org/project/2597fc3d-eb2c-411f-b02d-940c8347328d
```

File: lfx-mentorship/2023/01-Mar-May/README.md [hunk 5]
```diff
@@ -563,6 +609,18 @@ each subset of UnitedDeployment can have different metadata, container configura
 - Upstream Issue: https://github.com/oras-project/oras-www/issues/82
 - LFX URL: https://mentorship.lfx.linuxfoundation.org/project/7f633ade-64f5-477c-bcbe-7b6693329c63
 
+### Service Mesh Performance
+
+#### Adaptive Load Controller
```
Comment (nate-double-u) [inline lfx-mentorship/2023/01-Mar-May/README.md:614]:
```suggestion
#### Adaptive Load Controller II
```

File: lfx-mentorship/2023/01-Mar-May/README.md [hunk 6]
```diff
@@ -460,6 +483,29 @@ Note: This item is a work in progress. The selected mentee is expected to contin
...
 - LFX URL: https://mentorship.lfx.linuxfoundation.org/project/50cdbd65-e0cd-4c0f-8c63-6bd5c603ba89
 
+#### Meshery
+
+##### Distributed workflow engine
+
+- Description: Integrate a new architectural component into Meshery: a workflow engine. This project involves shifting Meshery off of bitcask and off of sqlite over to postgres using gorm (golang). Interns will familiarize with concepts of orchestration engines, including chaining workflows, and content lifecycle management.
+- Recommended Skills: Golang, Temporal, ReactJS
+- Mentor(s): Lee Calcote (@leecalcote, leecalcote@gmail.com), Ashish Tiwari (ashishjaitiwari15112000@gmail.com)
+- Issue: <https://github.com/meshery/meshery/issues/3934>
```
Comment (nate-double-u) [inline lfx-mentorship/2023/01-Mar-May/README.md:493]:
```suggestion
- Issue: https://github.com/meshery/meshery/issues/3934
- LFX URL: https://mentorship.lfx.linuxfoundation.org/project/73202d21-d4ca-4435-9a73-f326c9b3e796
```

File: lfx-mentorship/2023/01-Mar-May/README.md [hunk 7]
```diff
@@ -460,6 +483,29 @@ Note: This item is a work in progress. The selected mentee is expected to contin
...
+  
+##### Multi-user cloud native playground
+
+- Description: Advance the cloud native playground in which any CNCF project can be explored. Meshery’s genesis is that of helping teach people about cloud native technology and enabling to operate various types of cloud native infrastructure confidently. The proposed project is aimed at furthering this mission by infusing multi-user collaboration as a pervasisve feature so that users can learn together in a running instance of Meshery.
+- Recommended Skills: ReactJS, CSS, Golang (nice-to-have)
+- Mentor(s): Lee Calcote (@leecalcote, leecalcote@gmail.com), Abhishek Kumar (@abhishek-kumar09, abhimait1909@gmail.com)
+- Issue: <https://github.com/layer5io/meshery/issues/2931>
```
Comment (nate-double-u) [inline lfx-mentorship/2023/01-Mar-May/README.md:500]: We can't accept this project proposal as is -- the GitHub issue is linking to a private Layer 5 repo.

File: lfx-mentorship/2023/01-Mar-May/README.md [hunk 8]
```diff
@@ -460,6 +483,29 @@ Note: This item is a work in progress. The selected mentee is expected to contin
...
+
+#### Distributed client-side policy evaluation in WASM and Rego
+
+- Description: Meshery's highly dynamic infrastructure configuration capabilities require real-time evaluation of complex policies. Policies of various types and with a high number of parameters need to be evaluted client-side. With policies expressed in Rego, the goal of this project is to incorporate use of the https://github.com/open-policy-agent/golang-opa-wasm project into Meshery UI, so that a powerful, real-time user experience is possible.
+- Recommended Skills: Golang, Open Policy Agent, WebAssembly
+- Mentor(s): Lee Calcote (@leecalcote, leecalcote@gmail.com), Ashish Tiwari (ashishjaitiwari15112000@gmail.com)
+- Issue: <https://github.com/meshery/meshery/issues/7019>
```
Comment (nate-double-u) [inline lfx-mentorship/2023/01-Mar-May/README.md:507]:
```suggestion
- Issue: https://github.com/meshery/meshery/issues/7019
- LFX URL: https://mentorship.lfx.linuxfoundation.org/project/7e3382be-5d82-443e-b0bc-4dcd2194705d
```

**Review by nate-double-u — CHANGES_REQUESTED** [review]
Hi @leecalcote,

I've submitted the projects below except for one, please see the details inline. For the others I've added the LFX URLs, but we can't merge this till LFX approves them.

File: lfx-mentorship/2023/01-Mar-May/README.md (hunk 7 above)
Comment (leecalcote) [reply lfx-mentorship/2023/01-Mar-May/README.md:500]: Oh, this is an old reference to where the Meshery project was originally located, but post-donation was moved from `layer5io/meshery` to  `meshery/meshery`. Funky, that the GitHub redirection isn't working. 

I went ahead and created a fresh issue: https://github.com/meshery/meshery/issues/7020

Comment (leecalcote) [inline lfx-mentorship/2023/01-Mar-May/README.md:500]:
```suggestion
- Issue: <https://github.com/meshery/meshery/issues/7020>
```

File: lfx-mentorship/2023/01-Mar-May/README.md (hunk 5 above)
Comment (leecalcote) [reply lfx-mentorship/2023/01-Mar-May/README.md:614]: Thank you for tossing in this "II". 👍

File: lfx-mentorship/2023/01-Mar-May/README.md [hunk 9]
```diff
@@ -460,6 +484,31 @@ Note: This item is a work in progress. The selected mentee is expected to contin
...
+- Mentor(s): Lee Calcote (@leecalcote, leecalcote@gmail.com), Ashish Tiwari (ashishjaitiwari15112000@gmail.com)
+- Issue: https://github.com/meshery/meshery/issues/3934
+- LFX URL: https://mentorship.lfx.linuxfoundation.org/project/73202d21-d4ca-4435-9a73-f326c9b3e796
+  
+##### Multi-user cloud native playground
+
+- Description: Advance the cloud native playground in which any CNCF project can be explored. Meshery’s genesis is that of helping teach people about cloud native technology and enabling to operate various types of cloud native infrastructure confidently. The proposed project is aimed at furthering this mission by infusing multi-user collaboration as a pervasisve feature so that users can learn together in a running instance of Meshery.
+- Recommended Skills: ReactJS, CSS, Golang (nice-to-have)
+- Mentor(s): Lee Calcote (@leecalcote, leecalcote@gmail.com), Abhishek Kumar (@abhishek-kumar09, abhimait1909@gmail.com)
+- Issue: <https://github.com/meshery/meshery/issues/7020>
```
Comment (nate-double-u) [inline lfx-mentorship/2023/01-Mar-May/README.md:502]:
```suggestion
- Issue: https://github.com/meshery/meshery/issues/7020
- LFX URL: https://mentorship.lfx.linuxfoundation.org/project/2ee7a912-e26e-4602-9dfc-4febe3842df3
```

Comment (nate-double-u) [reply lfx-mentorship/2023/01-Mar-May/README.md:502]: @leecalcote thanks for updating the upstream issue!

**Review by nate-double-u — APPROVED** [review]
This all looks good now.

/hold merge till LFX approves

**Comment by nate-double-u** [comment]
/un-hold

LFX has approved, and the projects are open for applications.

**Pull request merged**